### PR TITLE
Runner: Move schedule loading and dir setup into module

### PR DESCRIPTION
**Pull request by kalikiana**
See: https://progress.opensuse.org/issues/81899

### CODE REVIEW

File: OpenQA/Isotovideo/Runner.pm [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+sub new ($class, %args) {
+    my $self = $class->SUPER::new(
+        casedir => path($args{casedir}),
+        needles_dir => $args{needles_dir},
+        productdir => path($args{productdir}),
+        test_git_hash => undef,
+    );
+
+    checkout_git_repo_and_branch('CASEDIR');
```
Comment (perlpunk) [inline OpenQA/Isotovideo/Runner.pm:29]: I would avoid doing too much in a constructor.

**Review by okurz — COMMENTED** [review]
I don't yet see the benefit of moving from one file to another as long as there is not less in isotovideo. As announced I am currently preparing a change to move more out of isotovideo into modules.

**Comment by kalikiana** [comment]
> I don't yet see the benefit of moving from one file to another as long as there is not less in isotovideo. As announced I am currently preparing a change to move more out of isotovideo into modules.

Which isn't my goal. I'm doing this to enable unit tests.

File: OpenQA/Isotovideo/Runner.pm [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+# a run is comprised of these individual resources
+has [qw(casedir needles_dir productdir test_git_hash)] => undef;
+
+sub new ($class, %args) {
+    my $self = $class->SUPER::new(
+        casedir => path($args{casedir}),
+        needles_dir => $args{needles_dir},
+        productdir => path($args{productdir}),
+        test_git_hash => undef,
+    );
```
Comment (Martchus) [inline OpenQA/Isotovideo/Runner.pm:27]:
```suggestion
sub new ($class, @args) {
    my $self = $class->SUPER::new(@args);
```

File: OpenQA/Isotovideo/Runner.pm (hunk 1 above)
Comment (Martchus) [reply OpenQA/Isotovideo/Runner.pm:29]: Avoiding this invocation directly in the constructor would mean we could avoid having the constructor altogether (which would be a plus, indeed).

**Review by Martchus — COMMENTED** [review]
I'm wondering whether it makes sense opening two construction sites very close to each other (considering @okurz changes to move things out of isotovideo). But I suppose the refactoring is generally ok if it helps to improve tests.

File: OpenQA/Isotovideo/Runner.pm (hunk 2 above)
Comment (kalikiana) [reply OpenQA/Isotovideo/Runner.pm]: That would mean the caller has to provide `path` objects, though.

**Pull request closed without merging**